doc/sphinx/practical-tools/utilities.rst

```diff
@@ -92,17 +92,55 @@ CoqMakefile
   is a makefile for ``GNU Make`` with targets to build the project
   (e.g. generate .vo or .html files from .v or compile .ml* files)
   and install it in the ``user-contrib`` directory where the |Coq|
-  library is installed. Run ``make`` with the ``-f CoqMakefile``
-  option to use ``CoqMakefile``.
+  library is installed.
 
 CoqMakefile.conf
   contains make variables assignments that reflect
   the contents of the ``_CoqProject`` file as well as the path relevant to
   |Coq|.
 
+The recommended approach is to embed ``CoqMakefile`` into a standard
+``Makefile`` of the following form:
+
+.. example::
+
+  ::
+
+      # KNOWNTARGETS will not be passed along to CoqMakefile
+      KNOWNTARGETS := CoqMakefile extra-stuff extra-stuff2
+      # KNOWNFILES will not get implicit targets from the final rule, and so
+      # depending on them won't invoke the submake
+      # Warning: These files get declared as PHONY, so any targets depending
+      # on them always get rebuilt
+      KNOWNFILES   := Makefile _CoqProject
+
+      .DEFAULT_GOAL := invoke-coqmakefile
+
+      CoqMakefile: Makefile _CoqProject
+              $(COQBIN)coq_makefile -f _CoqProject -o CoqMakefile
+
+      invoke-coqmakefile: CoqMakefile
+              $(MAKE) --no-print-directory -f CoqMakefile $(filter-out $(KNOWNTARGETS),$(MAKECMDGOALS))
+
+      .PHONY: invoke-coqmakefile $(KNOWNFILES)
+
+      ####################################################################
+      ##                      Your targets here                         ##
+      ####################################################################
+
+      # This should be the last rule, to handle any targets not declared above
+      %: invoke-coqmakefile
+              @true
+
+An alternative would have been to give instead option ``-o Makefile``
+to ``coq_makefile`` but the advantage of a wrapper is that it provides
+a way independent of the version of Coq to regenerate a ``Makefile``
+specific to the current version of Coq. Additionally, the master
+``Makefile`` can be extended with targets not specific to Coq.
 
 An optional file ``CoqMakefile.local`` can be provided by the user in order to
-extend ``CoqMakefile``. In particular one can declare custom actions to be
+extend ``CoqMakefile`` with other Coq targets.
```

Comment:
This is not correct. If you want to add a brand new target, you can just add it your wrapping makefile.
Makefile.local is to extend existing targets or set variables.

Reply:
I was rephrasing the text that followed: "Similarly one can customize the install target or even provide new targets." Does this sentence need to be changed also?

Reply:
OK, let me rephrase: "Makefile.local is mainly to extend existing targets or set variables. You can also add new targets".
The current text does not even mention what I consider to be the main use cases.

Reply:
@gares (and @JasonGross): I pushed an update but I'm not fully sure I expressed correctly what you intended. Don't hesitate to make a suggestion.